### PR TITLE
feature: update information on Kotlin/Wasm

### PR DESCRIPTION
**Pull request by koshachy**

### CODE REVIEW

File: docs/topics/wasm/wasm-overview.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[//]: # (title: Kotlin Wasm)
+
+> Kotlin Wasm is [Experimental](components-stability.md).
+> It may be changed at any time. Use it only for evaluation purposes.
+> 
+> We would appreciate your feedback on it in [YouTrack](https://kotl.in/issue).
+>
+{type="warning"}
+
+[WebAssembly (Wasm)](https://webassembly.org) is a binary format for a virtual machine that can run in web browsers.
```
Comment (bashor) [inline docs/topics/wasm/wasm-overview.md:10]: It's not browser only

Comment (bashor) [reply docs/topics/wasm/wasm-overview.md:10]: https://webassembly.org
> WebAssembly (abbreviated Wasm) is a binary instruction format for a stack-based virtual machine. Wasm is designed as a portable compilation target for programming languages, enabling deployment on the web for client and server applications.

Comment (bashor) [reply docs/topics/wasm/wasm-overview.md:10]: Also, you can get some inspiration from https://blog.jetbrains.com/kotlin/2023/04/kotlin-1-8-20-released/#wasm

File: docs/topics/wasm/wasm-overview.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Kotlin/Wasm is a new compilation target for Kotlin. 
+With Kotlin/Wasm, you can take advantage of the benefits of WebAssembly to write web applications that are capable of running on a variety of environments and devices.
+
+Kotlin features a new experimental Kotlin/Wasm compiler that provides several advantages:
+
+* Faster compilation speeds than the deprecated `wasm32` Kotlin/Native target since Kotlin/Wasm doesn't rely on LLVM.
+* Improved interoperability with JavaScript and better integration with browsers, thanks to the [Wasm garbage collection](https://github.com/WebAssembly/gc) feature.
+* Potentially faster application startup compared to Kotlin/JS and JavaScript because Wasm has a compact and easy-to-parse bytecode.
+* Improved application runtime performance compared to Kotlin/JS and JavaScript because Wasm is a statically typed language.
```
Comment (bashor) [inline docs/topics/wasm/wasm-overview.md:22]: Let's omit it now.

Comment (bashor) [reply docs/topics/wasm/wasm-overview.md:22]: It should be revisited.

File: docs/topics/wasm/wasm-overview.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+You can use Kotlin standard library (`stdlib`) and test library (`kotlin.test`) for Kotlin/Wasm out of the box.
+
+[Learn more about Kotlin/Wasm in our YouTube video](https://www.youtube.com/watch?v=-pqz9sKXatw).
```
Comment (bashor) [inline docs/topics/wasm/wasm-overview.md:26]: This video is a bit old. Let's omit it now and add a new one soon we get a new one.

**Review by bashor — CHANGES_REQUESTED** [review]

File: docs/topics/wasm/wasm-overview.md (hunk 1 above)
Comment (koshachy) [reply docs/topics/wasm/wasm-overview.md]: Ok, rephrased a bit.

File: docs/topics/wasm/wasm-overview.md (hunk 2 above)
Comment (bashor) [reply docs/topics/wasm/wasm-overview.md]: it isn't clear from GitHub UI but originally it was comment over 17-22 lines

File: docs/topics/wasm/wasm-overview.md (hunk 3 above)
Comment (bashor) [reply docs/topics/wasm/wasm-overview.md]: IDK, do you think it's suitable here? cc @dromanov-jb

File: docs/topics/wasm/wasm-overview.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+* Faster compilation speeds than the deprecated `wasm32` Kotlin/Native target since Kotlin/Wasm doesn't rely on LLVM.
+* Improved interoperability with JavaScript and better integration with browsers, thanks to the [Wasm garbage collection](https://github.com/WebAssembly/gc) feature.
+* Potentially faster application startup compared to Kotlin/JS and JavaScript because Wasm has a compact and easy-to-parse bytecode.
+
+You can use Kotlin standard library (`stdlib`) and test library (`kotlin.test`) for Kotlin/Wasm out of the box.
+
+## Browser support
+
+Almost all modern browsers already support WebAssembly 1.0.
```
Comment (bashor) [inline docs/topics/wasm/wasm-overview.md:27]: I prefer something like "All major browsers"

File: docs/topics/wasm/wasm-overview.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+To run applications built with Kotlin/Wasm in a browser, you need to enable an experimental [garbage collection feature](https://github.com/WebAssembly/gc).
+
+[Learn how to enable Kotlin/Wasm in Get started with Kotlin/Wasm](wasm-get-started.md#troubleshooting).
+
+## Interoperability
+
+Kotlin/Wasm allows you to both use JavaScript code from Kotlin and Kotlin code from JavaScript.
```
Comment (bashor) [inline docs/topics/wasm/wasm-overview.md:34]: "use JS code and Browser API"?

Comment (bashor) [reply docs/topics/wasm/wasm-overview.md:34]: Declarations for Browser APIs available out of the box

**Review by bashor — CHANGES_REQUESTED** [review]

**Comment by bashor** [comment]
@koshachy Let's fix these two and publish
https://github.com/JetBrains/kotlin-web-site/pull/3567#discussion_r1188751287
https://github.com/JetBrains/kotlin-web-site/pull/3567#discussion_r1188761583

**Pull request merged**